R/estimateAlpha.R

```diff
@@ -0,0 +1,274 @@
+#' Estimate alpha diversity indices.
+#' 
+#' The function estimates alpha diversity indices optionally using of rarefaction,
+#' then stores results at \code{\link{colData}}.
+#' 
+#' @param x a \code{\link{SummarizedExperiment}} object.
+#' 
+#' @param assay.type the name of the assay used for
+#'   calculation of the sample-wise estimates.
+#'   
+#' @param index a \code{character} vector, specifying the alpha diversity indices
+#'   to be calculated
+#'   
+#' @param name a name for the column(s) of the colData the results should be
+#'   stored in. By default this will use the original names of the calculated
+#'   indices.
+#'   
+#' @param ... optional arguments.
+#' 
+#' @param n.iter a single \code{integer} value for the number of rarefaction
+#' rounds.
+#' 
+#' @param rarefaction.depth a \code{double} value as for the minimim size or 
+#' rarefaction.depth. (default: \code{min(colSums(assay(x, "counts")), na.rm = TRUE)})
+#' 
```

Comment:
I think the default value should be NULL --> no rarefaction. It is more clear, I think

Reply:
Or mayne we could use TRUE/FALSE values and then numeric

TRUE --> default rarefsction with message that says the rarefaction depth

FALSE --> no rarefaction

Numeric --> use user-defined depth